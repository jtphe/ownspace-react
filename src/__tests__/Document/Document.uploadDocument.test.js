import React from 'react';
import UploadDocument from '@components/Document/UploadDocument';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { View } from 'react-native';
import { CLIENT_COLOR_PRIMARY } from '@constants';

describe('UploadDocument component tests', () => {
  let wrapper;
  const text = 'Loading to test';

  beforeEach(() => {
    wrapper = shallow(<UploadDocument content={text} />);
  });

  it('UploadDocument snapshot', async () => {
    let component;
    await renderer.act(async () => {
      component = renderer.create(<UploadDocument />);
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Icon loading', () => {
    const iconLoading = wrapper.find(View).get(0).props.children[0].props
      .children[0];

    expect(iconLoading.props.loading).toEqual(true);
    expect(iconLoading.props.style).toEqual({ marginRight: -20 });
    expect(iconLoading.props.color).toEqual(CLIENT_COLOR_PRIMARY);
    expect(iconLoading.props.uppercase).toEqual(false);
  });

  it('Text loading test', () => {
    const textLoading = wrapper.find(View).get(0).props.children[0].props
      .children[1].props.children;
    expect(textLoading[0].props.children).toEqual(text);
  });
});

Enzyme.configure({ adapter: new Adapter() });
