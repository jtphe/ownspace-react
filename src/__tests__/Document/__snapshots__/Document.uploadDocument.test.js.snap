// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadDocument component tests UploadDocument snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "marginLeft": 20,
        "paddingBottom": 12,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 0,
          "marginRight": -20,
          "minWidth": 64,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "marginLeft": 12,
                  "marginRight": -4,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "height": 16,
                  "opacity": 1,
                  "width": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 16,
                      "transform": Array [
                        Object {
                          "rotate": "45deg",
                        },
                      ],
                      "width": 16,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 8,
                        "overflow": "hidden",
                        "width": 16,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "height": 16,
                          "transform": Array [
                            Object {
                              "translateY": 0,
                            },
                            Object {
                              "rotate": "-165deg",
                            },
                          ],
                          "width": 16,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 8,
                            "overflow": "hidden",
                            "width": 16,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "borderColor": "#003466",
                              "borderRadius": 8,
                              "borderWidth": 1.6,
                              "height": 16,
                              "width": 16,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 16,
                      "transform": Array [
                        Object {
                          "rotate": "45deg",
                        },
                      ],
                      "width": 16,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 8,
                        "overflow": "hidden",
                        "top": 8,
                        "width": 16,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "height": 16,
                          "transform": Array [
                            Object {
                              "translateY": -8,
                            },
                            Object {
                              "rotate": "345deg",
                            },
                          ],
                          "width": 16,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 8,
                            "overflow": "hidden",
                            "width": 16,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "borderColor": "#003466",
                              "borderRadius": 8,
                              "borderWidth": 1.6,
                              "height": 16,
                              "width": 16,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  false,
                  Object {
                    "color": "#003466",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "width": 240,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        selectable={true}
        style={
          Array [
            Object {
              "flexWrap": "wrap",
              "paddingHorizontal": 12,
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          selectable={true}
          style={
            Array [
              Object {
                "fontSize": 10,
                "paddingLeft": 13,
              },
              Object {
                "fontFamily": "HelveticaNeue",
              },
            ]
          }
        >
          July 1, 2020 5:09 PM
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 0.4,
        "marginBottom": 12,
        "marginLeft": 75,
      }
    }
  />
</View>
`;
